integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkCredentialRenewalTest.java

```diff
@@ -74,37 +100,59 @@ public static void beforeAll() {
             .config(
                 "spark.sql.catalog.spark_catalog",
                 "org.apache.spark.sql.delta.catalog.DeltaCatalog")
-            .config(testCatalogKey, UCSingleCatalog.class.getName())
-            .config(testCatalogKey + ".uri", SERVER_URL)
-            .config(testCatalogKey + ".token", AUTH_TOKEN)
-            .config(testCatalogKey + ".warehouse", CATALOG_NAME)
-            .config(testCatalogKey + ".renewCredential.enabled", String.valueOf(RENEW_CRED_ENABLED))
-            .getOrCreate();
-
-    sql("CREATE SCHEMA %s.%s", CATALOG_NAME, SCHEMA_NAME);
-    sql(
-        "CREATE TABLE %s (id INT) USING delta LOCATION '%s/%s' PARTITIONED BY (partition INT)",
-        SRC_TABLE, S3_BASE_LOCATION, UUID.randomUUID());
-    sql(
-        "CREATE TABLE %s (id INT) USING delta LOCATION '%s/%s'",
-        DST_TABLE, S3_BASE_LOCATION, UUID.randomUUID());
+            .config("spark.sql.adaptive.enabled", "false")
+            .config("spark.sql.adaptive.coalescePartitions.enabled", "false");
+
+    String testCatalogKey = String.format("spark.sql.catalog.%s", CATALOG_NAME);
+    // Use the UCSingleCatalog by default.
+    builder.config(testCatalogKey, UCSingleCatalog.class.getName());
+    // Set all the specified catalog props.
+    catalogProps.forEach((k, v) -> builder.config(testCatalogKey + "." + k, v));
+    spark = builder.getOrCreate();
+
+    sql("CREATE SCHEMA IF NOT EXISTS %s.%s", CATALOG_NAME, SCHEMA_NAME);
+    if (managedTable) {
+      // TODO: SQL-managed table creation is not supported yet, so we create the table manually.
+      // Once SQL support is added, use the SQL to create tables.
+      sql("DELETE FROM %s", srcTable(true));
+      sql("DELETE FROM %s", dstTable(true));
+    } else {
+      sql("DROP TABLE IF EXISTS %s", srcTable(false));
+      sql("DROP TABLE IF EXISTS %s", dstTable(false));
+      sql(
+          "CREATE TABLE %s (id INT) USING delta LOCATION '%s/%s' PARTITIONED BY (partition INT)",
+          srcTable(false), baseLocation, UUID.randomUUID());
+      sql(
+          "CREATE TABLE %s (id INT) USING delta LOCATION '%s/%s'",
+          dstTable(false), baseLocation, UUID.randomUUID());
+    }
   }
 
-  @AfterAll
-  public static void afterAll() {
-    sql("DROP TABLE IF EXISTS %s", SRC_TABLE);
-    sql("DROP TABLE IF EXISTS %s", DST_TABLE);
-    sql("DROP SCHEMA IF EXISTS %s.%s", CATALOG_NAME, SCHEMA_NAME);
-    spark.stop();
+  @AfterEach
+  public void afterEach() {
+    sql("DROP TABLE IF EXISTS %s", srcTable(false));
+    sql("DROP TABLE IF EXISTS %s", dstTable(false));
+
+    if (spark != null) {
+      spark.stop();
+      spark = null;
+    }
   }
 
-  @Test
-  public void testLongRunningJob() {
+  @ParameterizedTest
+  @MethodSource("credentialRenewalArguments")
+  public void testLongRunningJob(
+      boolean managedTable, String baseLocation, Map<String, String> catalogProps) {
```

Comment:
isManagedTable

more obvious that its a boolean.

Reply:
Thanks for the comment,  let me refine them and make them more clear.